python/social_bot/envs/grocery_ground.py

```diff
@@ -62,6 +62,12 @@ def __init__(self, reward_shaping=False, random_goal=False, **kwargs):
         ]
         self.task_vocab = self.task_vocab + self._objects_in_world + self._objects_to_insert
 
+    def run(self, agent, world):
+        if self._random_goal:
+            random_id = random.randrange(len(self._objects_to_insert))
+            self.set_goal_name(self._objects_to_insert[random_id])
+        yield super(GroceryGroundGoalTask, self).run(agent, world)
```

Comment:
Did you test this? I think the right way to do this is: yield from super(...).run(...). See https://docs.python.org/3/whatsnew/3.3.html#pep-380

Reply:
And we can use the simpler new syntax: super().run(agent, world)

Reply:
My mistake. Should use "yield from" here.